src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java

```diff
@@ -1118,6 +1118,45 @@ public Collection<Experiment> findExperimentsByGenericMetaDataSearch(String quer
 
 	}
 
+	public Collection<Experiment> findExperimentsByProtocolCodeName(String codeName, List<String> projects)
```

Comment:
New service which will handle filter the experiments like some of our other services already do it.